src/Sulu/Bundle/AdminBundle/Controller/IconController.php

```diff
@@ -0,0 +1,171 @@
+        $search = $request->query->get('search');
+        $provider = $iconSet[0];
+        $path = $iconSet[1] ?? '';
+
+        // Get icons depending on the provider type.
+        switch ($provider) {
+            case 'svg':
+                $icons = $this->getIconsAsSVGs($path);
+                break;
+            case 'icomoon':
+                $icons = $this->getIconsAsIcomoon($path);
+                break;
+            default:
+                throw new BadRequestException();
+        }
+
+        // Implement a simple search functionality.
+        if ($search) {
+            $filteredIcons = [];
+
+            foreach ($icons as $icon) {
+                if (\str_contains($icon['id'], $search)) {
+                    $filteredIcons[] = $icon;
+                }
+            }
+
+            $icons = $filteredIcons;
+        }
+
+        // Sort by ID.
+        \usort($icons, fn ($a, $b) => $a['id'] <=> $b['id']);
+
+        return $this->handleView(
+            $this->view(
+                new CollectionRepresentation(
+                    $icons,
+                    'icons'
+                )
+            )
+        );
+    }
+
+    /**
+     * Return a set of SVG icons from a specific directory.
+     *
+     * @return array<array{id: string, content: string}>
+     */
+    public function getIconsAsSVGs(string $path)
+    {
+        $cacheKey = 'icons_' . \md5($path);
+
+        return $this->cache->get($cacheKey, function() use ($path) {
+            $icons = [];
+
+            $finder = new Finder();
+            $finder->in($path);
+
+            foreach ($finder as $file) {
+                $icons[] = [
+                    'id' => $file->getBasename('.svg'),
+                    'content' => $file->getContents(),
+                ];
+            }
+
+            return $icons;
+        });
+    }
+
+    /**
+     * Return a set of icons from the icomoon selection.json.
+     *
+     * @return array<array{id: string, content: string}>
+     */
+    public function getIconsAsIcomoon(string $selectionJsonPath)
+    {
+        $icons = [];
+        $selectionContent = \file_get_contents($selectionJsonPath);
+
+        if (!$selectionContent) {
+            return $icons;
+        }
+
+        $selection = (array) \json_decode($selectionContent);
+        $iconsArray = (array) $selection['icons'];
+
+        /**
+         * @var \stdClass $icon
+         */
+        foreach ($iconsArray as $icon) {
+            $paths = [];
+
+            \assert(isset($icon->icon) && \is_object($icon->icon));
+            \assert(isset($icon->icon->paths) && \is_array($icon->icon->paths));
+            \assert(isset($icon->properties) && \is_object($icon->properties));
+            \assert(isset($icon->properties->name) && \is_string($icon->properties->name));
+
+            /**
+             * @var string $path
+             */
+            foreach ($icon->icon->paths as $index => $path) {
```

Comment:
This does not support multicolor svgs from icomoon they have `fill` with different colors